blog/content/post/2022-12-22-fobnail_application.md

```diff
@@ -0,0 +1,363 @@
+allows the Fobnail owner to verify the trustworthiness of the running system
+before performing any sensitive operation. This project was founded by [NlNet
+Foundation](https://nlnet.nl/). More information about the project can be found
+in the [Fobnail documentation](https://fobnail.3mdeb.com/). Also, make sure to
+read other posts related to this project by visiting
+[fobnail](https://blog.3mdeb.com/tags/fobnail/) tag.
+
+# Scope of current phase
+
+This phase focused on using Fobnail in a real-life use case, namely using it to
+access LUKS2 decryption key if and only if the PCR measurements are valid. In
+this post only high-level usage will be described, if you want to know what
+happens under the hood or want to modify it to your needs see
+[documentation](https://fobnail.3mdeb.com/examples/disk_encryption/).
+
+# Changes
+
+In this phase we completely switched the architecture around. Now the CoAP
+server is located on Fobnail Token, instead of each of PC applications. This
+way we could implement [API endpoints](https://fobnail.3mdeb.com/fobnail-api/),
+including [Fobnail Token Services (FTS)](https://fobnail.3mdeb.com/fobnail-api/#fobnail-token-services),
+which allows for more elaborate applications without having to re-flash Token
+firmware each time.
+
+As of now, only storage services are fully defined in FTS. Few other endpoints
+for working with cryptographic keys were listed, but they are yet to be
+implemented. There is also an endpoint `/api/version` which reports all versions
+of API supported by Token - the version will be increased only when
+backward-incompatible changes are done.
+
+There are now two separate Attester applications - one with provisioning code,
+and one without. Former should be used just for initial provisioning, and the
+latter for normal use. We hope that by simplifying more frequently used form of
+Attester (that is, the application used just for attestation, not provisioning)
+we can reduce the surface of attack. It also helps with rare cases when users
+inadvertently plugged in unprovisioned Token - in that case Token would also
+lit _all-good_ green LED in the end, making it indistinguishable from normal
+attestation. For the same reason we also gave up on the idea of unprovisioning
+the token by holding the button - neighboring USB device could potentially keep
+the button pressed
+
+Speaking of LEDs, we now have to steer them in a non-blocking manner. Simple
+delay loops were good enough when the Token was CoAP client, but now it has to
+be able to respond to the clients without making them wait too long, especially
+for longer signals like attestation result. We took advantage of that forced
+change to [expand and standardize **TBD**](TBD) blink codes produced by Fobnail
+Token.
+
+## Building and running
+
+Building hasn't change much since [last time](https://blog.3mdeb.com/2022/2022-05-25-fobnail_provisioning/#building-and-running).
```

Comment:
Shouldn't official building instruction be in upstream documentation instead of blog post? Why we linking blog post?

Reply:
> Because we don't have versioned documentation. Build instructions used to change with each release.

@krystian-hebel there are other ways to solve problem of stable instructions. I see no issue with keeping in documentation building steps for various versions (we do that already for [Dasharo (coreboot+SeaBIOS) vs (coreboot+UEFI)](https://docs.dasharo.com/variants/dell_optiplex/building-manual/#install-dasharo-firmware)). I don't understand point of versioned documentation, whole documentation (mkdocs) is tracked in git. OTOH there is no need for documentation versioning and asking contributors for switching documentation version (or whatever solution you had in mind) according to version. Also likelihood of compiling some old version for project which didn't reach mass usability and keeping documentation (despite it is tracked github), not saying about providing support for old versions of software, is unnecessary redundant and we could without concerns overwrite old building steps and use DRY by sending contributors to the documentation.

I will review MR with documentation update ASAP.

Reply:
Added link to documentation in https://github.com/3mdeb/news-and-ideas/pull/504/commits/289e3a8aa2a8c87fe55bb018ddfccbac0e03a6f3. I've left link to previous post for SEO.